modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/query/builder/FilterQueries.scala

```diff
@@ -0,0 +1,223 @@
+package ch.epfl.bluebrain.nexus.kg.indexing.query.builder
+
+import akka.http.scaladsl.model.Uri
+import cats.instances.string._
+import cats.syntax.show._
+import ch.epfl.bluebrain.nexus.common.types.Version
+import ch.epfl.bluebrain.nexus.kg.core.domains.DomainId
+import ch.epfl.bluebrain.nexus.kg.core.organizations.OrgId
+import ch.epfl.bluebrain.nexus.kg.core.schemas.SchemaId
+import ch.epfl.bluebrain.nexus.kg.indexing.Qualifier._
+import ch.epfl.bluebrain.nexus.kg.indexing.filtering.Expr.{ComparisonExpr, LogicalExpr, NoopExpr}
+import ch.epfl.bluebrain.nexus.kg.indexing.filtering.Op.{And, Eq}
+import ch.epfl.bluebrain.nexus.kg.indexing.filtering.Term.{LiteralTerm, UriTerm}
+import ch.epfl.bluebrain.nexus.kg.indexing.filtering.{Expr, Filter, Op}
+import ch.epfl.bluebrain.nexus.kg.indexing.pagination.Pagination
+import ch.epfl.bluebrain.nexus.kg.indexing.query.builder.FilterQueries._
+import ch.epfl.bluebrain.nexus.kg.indexing.query.{QueryResults, QuerySettings, SparqlQuery}
+import ch.epfl.bluebrain.nexus.kg.indexing.{ConfiguredQualifier, Qualifier}
+
+/**
+  * Collection of queries.
+  *
+  * @param queryClient   the client to use for executing the queries
+  * @param querySettings the default query settings
+  * @tparam F  the monadic effect type
+  * @tparam Id the generic type which defines the response's payload
+  */
+class FilterQueries[F[_], Id](queryClient: SparqlQuery[F], querySettings: QuerySettings) {
```

Comment:
Once we unify indexing under a single namespace we can move the `Id` tparam to each method instead to be able to share the same FilterQueries instance.